### ISSUE TITLE
scipy.linalg.pinv2 default cutoff possibly too conservative

### ISSUE THREAD

**Issue by yaroslavvb** [scipy/scipy#10879]
pinv2 has the following calculation of threshold for small singular values

```
    if cond in [None,-1]:
        t = u.dtype.char.lower()
        factor = {'f': 1E3, 'd': 1E6}
        cond = factor[t] * np.finfo(t).eps

    rank = np.sum(s > cond * np.max(s))
```

I'm wondering if there's a reason for `1E3` and `1E6` values. This makes threshold significantly larger than default from Numerical Recipes
<img width="730" alt="Screenshot 2019-09-27 21 05 26" src="https://user-images.githubusercontent.com/23068/65811287-e7029580-e16a-11e9-89c4-87662f60a21e.png">

**Comment by ilayn** [comment]
Hi @yaroslavvb. Thank you for taking the time to report it. I think this is fixed already in #10067. Would you agree?

**Comment by yaroslavvb** [comment]
Aha! I should've checked the master. Now it seems to be in line with convention
